scripts/pgsql_functions.sql

```diff
@@ -256,3 +256,58 @@ BEGIN
     END LOOP;
 END;
 $$ LANGUAGE plpgsql;
+
+-- ============================================================= --
+
+-- FUNC DEF:
+CREATE OR REPLACE FUNCTION create_services_and_plans(
+    num_services INTEGER,
+    service_broker_id INTEGER,
+    num_service_plans INTEGER,
+    service_plan_public BOOLEAN,
+    num_visible_orgs INTEGER
+) RETURNS void AS
+$$
+DECLARE
+    service_guid TEXT;
+    service_label_prefix TEXT := 'perf-service-';
+    service_description_prefix TEXT := 'perf-service-description-';
+    service_bindable BOOLEAN := true;
+    service_plan_guid TEXT;
+    service_plan_name_prefix TEXT := 'perf-service-plan';
+    service_plan_description_prefix TEXT := 'perf-service-plan-description-';
+    service_plan_free BOOLEAN := true;
+    latest_service_id INTEGER;
+    latest_service_plan_id INTEGER;
+
+BEGIN
+    FOR _ IN 1..num_services LOOP
+        service_guid := gen_random_uuid();
+        INSERT INTO services (guid, label, description, bindable, service_broker_id)
+            VALUES (
+                service_guid,
+                service_label_prefix || service_guid,
+                service_description_prefix || service_guid,
+                service_bindable,
+                service_broker_id
+                ) RETURNING id INTO latest_service_id;
+        FOR _ IN 1..num_service_plans LOOP
+            service_plan_guid := gen_random_uuid();
+            INSERT INTO service_plans (guid, name, description, free, service_id, unique_id, public)
+                VALUES (
+                       service_plan_guid,
+                       service_plan_name_prefix || service_plan_guid,
+                       service_plan_description_prefix || service_plan_guid,
+                       service_plan_free,
+                       latest_service_id,
+                       'unique-' || service_plan_guid,
+                       service_plan_public
+                   ) RETURNING id INTO latest_service_plan_id;
+            INSERT INTO service_plan_visibilities (guid, service_plan_id, organization_id)
+                SELECT gen_random_uuid(), latest_service_plan_id, id
```

Comment:
Why exactly would you prefer  `SELECT ... INTO`? performance?

Reply:
I find it easier to read/understand and it would be consistent with other functions. We have something similar here: https://github.com/cloudfoundry-incubator/cf-performance-tests/blob/6043230548b70bfaae156cc7b631785c20456533/scripts/pgsql_functions.sql#L212-L213

Reply:
For us your approach seems hard to apply, we select multiple values at once, we think it is easier to leave it as it is (with  a subselect)

Reply:
Okay